### ISSUE TITLE
Subtypes of string passed as the webtest user are not handled in check_environ

### ISSUE THREAD

**Issue by tolomea** [django-webtest/django-webtest#50]
This came up when a section of test code resulted in a Django safe string being passed in as the username.
Webtest check_environ checks explicitly for an exact type match not a subtype match.
The relevant webtest lines are:

METADATA_TYPE = PY3 and (str, binary_type) or (str,)

assert type(environ[key]) in METADATA_TYPE, (
            "Environmental variable %s is not a string: %r (value: %r)"
            % (key, type(environ[key]), environ[key]))

Is there a reason why this isn't assert isinstance(type(eviron[key]), six.string_types) ?

Also this only came up on our Py3 build, digging into that led me to compat.py in django-webtest which Py2 effectively casts the safe string to a regular string:

    def to_string(s):
        return str(s)

But on Py3 leaves it as a safe string:

    def to_string(s):
        if isinstance(s, str):
            return s
        return str(s, 'latin1')

In general I'm not at all sure what that compat file is doing or why.

**Comment by tolomea** [comment]
cross posted https://github.com/Pylons/webtest/issues/158

**Comment by bertjwregeer** [comment]
https://github.com/Pylons/webtest/issues/158#issuecomment-466842437

tl;dr: WSGI environment PEP's specify only `str` types are allowed with characters in the `latin-1` character set.
